pxr/usd/usdSkel/validators.cpp

```diff
@@ -0,0 +1,89 @@
+//
+// Copyright 2024 Pixar
+//
+// Licensed under the terms set forth in the LICENSE.txt file available at
+// https://openusd.org/license.
+//
+#include "pxr/usd/usd/prim.h"
+#include "pxr/usd/usd/schemaRegistry.h"
+#include "pxr/usd/usd/validationError.h"
+#include "pxr/usd/usd/validationRegistry.h"
+#include "pxr/usd/usd/validator.h"
+#include "pxr/base/tf/token.h"
+#include "pxr/base/tf/stringUtils.h"
+#include "pxr/usd/usdSkel/validatorTokens.h"
+#include "pxr/usd/usdSkel/bindingAPI.h"
+
+#include <algorithm>
+#include <unordered_map>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+static
+UsdValidationErrorVector
+_SkelBindingApiAppliedValidator(const UsdPrim &usdPrim)
+{
+    UsdValidationErrorVector errors;
+
+    if (!usdPrim.HasAPI<UsdSkelBindingAPI>()){
+        std::vector<TfToken> primPropertyNames = usdPrim.GetPropertyNames();
+        static const std::unordered_set<TfToken, TfHash> skelPropertyNames = []() {
+            UsdSchemaRegistry& usdSchemaRegistry = UsdSchemaRegistry::GetInstance();
+            std::unique_ptr<UsdPrimDefinition> primDef = usdSchemaRegistry.BuildComposedPrimDefinition(TfToken(), {TfToken("SkelBindingAPI")});
```

Comment:
Ahh apologies, I should have noticed this before, you can use `UsdSkelTokens->SkelBindingAPI`, instead of `TfToken("SkelBindingAPI")`. Such tokens are often generated along with the schema.

https://github.com/PixarAnimationStudios/OpenUSD/blob/release/pxr/usd/usdSkel/tokens.cpp#L55